packages/presence-signal/README.md

```diff
@@ -0,0 +1,173 @@
+# @solid-primitives/presence-signal
+
+## [Example](https://stackblitz.com/edit/solidjs-templates-8eqpxz?file=src%2FApp.tsx)
+
+A small SolidJS utility to animate the presence of an element. Inspired by & directly forked from [`use-presence`](https://www.npmjs.com/package/use-presence).
+
+## The problem
+
+There are two problems that you have to solve when animating the presence of an element:
+
+1. During enter animations, you have to render an initial state where the element is hidden and only after the latest state has propagated to the DOM, you can can animate the final state that the element should animate towards.
+2. Exit animations are a bit tricky in SolidJS, since this typically means that a component unmounts. However when the component has already unmounted, you can't animate it anymore. A workaround is often to keep the element mounted, but that keeps unnecessary elements around and can hurt accessibility, as hidden interactive elements might still be focusable.
+
+## This solution
+
+This utility provides a lightweight solution where the animating element is only mounted the minimum of time, while making sure the animation is fully visible to the user. The rendering is left to the user to support all kinds of styling solutions.
+
+## Example
+
+```jsx
+const FirstExample = () => {
+  const [showStuff, setShowStuff] = createSignal(true);
+  const { isVisible, isMounted } = createPresenceSignal(showStuff, {
```

Comment:
I'm not sure if the name should include the `Signal` suffix. It's not a direct abstraction over `createSignal`, it uses more primitives inside than just a single signal, so it's creating more than that. Also, signals can be created anywhere and they don't hook into the current reactive owner in any way, they don't have to be disposed of using the ownership graph, but just garbage collected. `createPresenceSignal` is creating computations inside, so the place where it's used matters.
It could be just called `createPresence` and `createSwitchPresence`.

Reply:
I like this and think it fits the conventions already applied in the other packages in this repo. As a standalone package, these names made sense to me, but inside of the ecosystem of these packages I think the simpler name reads better.

Updated 👍